repositories/settings/api_test.go

```diff
@@ -76,23 +91,34 @@ func TestRepositoryFromReader(t *testing.T) {
 
 	t.Run("fails parsing equals signs", func(t *testing.T) {
 		fileMod := &fnFileMod{
-			fnGetFileReader: func(fileName string) io.Reader {
+			fnGetReadCloser: func(fileName string) (io.ReadCloser, error) {
 				if fileName != "settings.config" {
-					return nil
+					return nil, errors.New(invalidFileNameError)
 				}
-				return strings.NewReader(strings.Join([]string{
+				reader := strings.NewReader(strings.Join([]string{
 					"resolutionX=100",
 					"fov=60=60",
 				}, "\n"))
+				return &fnReadCloser{
+					fnRead: func(p []byte) (int, error) {
+						return reader.Read(p)
+					},
+					fnClose: func() error {
+						return nil
+					},
+				}, nil
 			},
 		}
 		expect := settings.ErrParseSyntax
 		var expectAs *settings.ErrParse
 		expectLine := 2
-		settingsMod := settings.New(fileMod)
-		reader := fileMod.GetFileReader("settings.config")
+		settingsMod := settings.New()
+		readerCloser, err := fileMod.GetReadCloser("settings.config")
```

Comment:
These file mod doesn't need to exist in these tests anymore.
Because the settings mod doesn't depend on the file mod anymore, you don't need to stub the file mod at all.
Just call `SetFromReader` with an `io.ReadCloser`.